### DISCUSSION TITLE
[QUESTION] how to return multiple errors?

### DISCUSSION THREAD

**Discussion by leognmotta** [hasura/graphql-engine#5666] [Question]
I am using action to make form validation from my API with AdonisJS 5, some times form validations can result in multiple errors, from hasura actions docs errors are returned as object in the api:

[https://hasura.io/docs/1.0/graphql/manual/actions/action-handlers.html](url)

```javascript
{
  "message": "<mandatory-error-message>",
  "code": "<optional-error-code>"
}
```
and hasura returns it as an array:

```javascript
{
  "errors": [
    {
      "extensions": {
        "path": "$",
        "code": "VALIDATION_ERROR"
      },
      "message": "password minLength validation failed"
    }
  ]
}
```

I tried many thinds already, but hasura expect message to be an object and the response to be an object not array. How can I improve user experience, showing all fields with errors? I am doing some gotacha, which is not something I want to do:

```javascript
    try {
        ...
    } catch (error) {
      response.status(400).json({
        code: 'VALIDATION_ERROR',
        message: JSON.stringify(error.messages.errors),
      })
    }
```
This way I can parse all errors and show to the user, but I really think there should be a better way, it just does not make sense for me returning errors as an array if I can only return I single error at once.

My final result:

```javascript
{
  "errors": [
    {
      "extensions": {
        "path": "$",
        "code": "VALIDATION_ERROR"
      },
      "message": "[{\"rule\":\"minLength\",\"field\":\"input.data.password\",\"message\":\"minLength validation failed\",\"args\":{\"minLength\":8}},{\"rule\":\"regex\",\"field\":\"input.data.password\",\"message\":\"regex validation failed\"}]"
    }
  ]
}
```

**Comment by jacdx** [comment]
There's an open PR for allowing extensions to be passed in the error obj that would give more flexibility, but unfortunately it's been stale for months. https://github.com/hasura/graphql-engine/pull/4351
